Content.Server/ShipEvent/Systems/ShipEventFactionSystem.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System.Linq;
+using Content.Server.Actions;
+using Content.Server.Chat.Managers;
+using Content.Server.Chat.Systems;
+using Content.Server.Ghost.Roles.Events;
+using Content.Server.IdentityManagement;
+using Content.Server.Mind.Components;
+using Content.Server.Roles;
+using Content.Server.ShipEvent.Components;
+using Content.Server.Shuttles.Components;
+using Content.Shared.Actions;
+using Content.Shared.Toggleable;
+
+namespace Content.Server.ShipEvent.Systems;
+
+public sealed class ShipEventFactionSystem : EntitySystem
+{
+	[Dependency] private readonly IEntityManager entMan = default!;
+	[Dependency] private readonly ChatSystem chatSystem = default!;
+    [Dependency] private readonly IChatManager chatMan = default!;
+    [Dependency] private readonly IdentitySystem identitySystem = default!;
+    [Dependency] private readonly ActionsSystem actionsSystem = default!;
+
+    private const float TeamCheckInterval = 5;
+    private float TeamCheckTimer = 0;
+
+	private int LastTeamNumber;
+
+	/// <summary>
+	/// Dictionary with all the teams, where spawner uid is the key, and faction is the value
+	/// </summary>
+	public Dictionary<EntityUid, PlayerFaction> Teams => teams;
+
+	private Dictionary<EntityUid, PlayerFaction> teams = new();
+
+    private Dictionary<EntityUid, string> shipNames = new();
+
+    public override void Initialize()
+	{
+		base.Initialize();
+		SubscribeLocalEvent<ShipEventFactionComponent, GhostRoleSpawnerUsedEvent>(OnSpawn);
+        SubscribeLocalEvent<ShipEventFactionViewComponent, ToggleActionEvent>(OnView);
+        SubscribeLocalEvent<ShipEventFactionViewComponent, ComponentInit>(OnViewInit);
+    }
+
+	public override void Update(float frametime)
+    {
+        if (TeamCheckTimer < TeamCheckInterval)
+        {
+            TeamCheckTimer += frametime;
+            return;
+        }
+        else
+        {
+            TeamCheckTimer -= TeamCheckInterval;
+            CheckTeams();
+        }
```

Comment:
```suggestion
        TeamCheckTimer = 0;
        CheckTeams();
```
Ладно по мелоче только напишу)